framework/PageInputs/HiddenField.tsx

```diff
@@ -0,0 +1,81 @@
+import { Button, InputGroup, TextInput } from '@patternfly/react-core';
+import React, { isValidElement } from 'react';
+import { useTranslation } from 'react-i18next';
+import { PageFormGroup } from '../PageForm/Inputs/PageFormGroup';
+
+/**
+ * Props for the HiddenField component.
+ * @interface IProps
+ */
+interface IProps {
+  /**
+   * Indicates if the value of the field is hidden.
+   * When true, the field value is masked and a clear button is displayed.
+   * When false, children components are displayed for user interaction.
+   */
+  isValueHidden: boolean;
+
+  /**
+   * Callback function triggered when the clear button is clicked.
+   * It is used to reset the field value to null or empty.
+   */
+  onClear: () => void;
+
+  /**
+   * The components to be displayed when the field value is not hidden,
+   * typically used for user interaction to input or edit the value.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Optional text that provides additional information or instructions
+   * for the field, displayed alongside the label.
+   */
+  labelHelp?: string;
+
+  /**
+   * The text label displayed above the field to indicate what
+   * information the field contains or expects.
+   */
+  label?: string;
+}
+
+interface ChildrenProps {
+  labelHelp?: string;
+  label?: string;
+}
+
+export function HiddenField({ onClear, isValueHidden, label, labelHelp, children }: IProps) {
+  const { t } = useTranslation();
+
+  let childrenProps: ChildrenProps = { label: '', labelHelp: '' };
+  if (isValidElement(children)) {
+    childrenProps = children.props as ChildrenProps;
+  }
+
+  const fieldLabel = label || childrenProps.label || '';
+  const fieldLabelHelp = labelHelp || childrenProps.labelHelp || '';
+
+  if (!isValueHidden) {
+    return <>{children}</>;
+  }
+
+  return (
+    <PageFormGroup label={fieldLabel} labelHelp={fieldLabelHelp}>
+      <InputGroup>
+        <TextInput
+          aria-label={t('hidden value')}
+          placeholder="••••••••••••••••••••••"
+          type="password"
+          autoComplete="off"
+          isDisabled={isValueHidden}
+        />
+        {isValueHidden && (
+          <Button variant="control" onClick={() => onClear()}>
```

Comment:
clear should not be enabled if ther is no value

Reply:
@himdel taking a look. Perhaps, renaming the variable, and making sure we have unit tests.

Reply:
@himdel update the logic. And there are unit tests covering both cases.